index.js

```diff
@@ -4,13 +4,33 @@ const accepts = require('accepts')
 
 const fp = require('fastify-plugin')
 
+const acceptsObjectSymbol = Symbol('acceptsObject')
+
+const methodNames = [
+  'charset', 'charsets',
+  'encoding', 'encodings',
+  'language', 'languages',
+  'type', 'types'
+]
+
 function acceptsMethod () {
-  return accepts(this.req)
+  if (!this.req[acceptsObjectSymbol]) {
+    this.req[acceptsObjectSymbol] = accepts(this.req)
+  }
+
+  return this.req[acceptsObjectSymbol]
 }
 
 function fastifyAcceptHeader (fastify, options, done) {
   fastify.decorateRequest('accepts', acceptsMethod)
 
+  methodNames.forEach(methodName => {
+    fastify.decorateRequest(methodName, function () {
+      const acceptsObject = this.accepts()
+      return acceptsObject[methodName].apply(acceptsObject, arguments)
```

Comment:
leaky arguments is horribly slow in every Node versions released so far. Given that all methods gets very few paramenters, I would actually call the fuction with the parameters directly.